File: src/test/java/com/example/dpop/orchestrator/ProcessControllerIntegrationTest.java
package com.example.dpop.orchestrator;

import org.junit.jupiter.api.Test;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.test.web.server.LocalServerPort;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.web.client.RestTemplate;

import static org.assertj.core.api.Assertions.assertThat;

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
@ActiveProfiles("test")
class ProcessControllerIntegrationTest {

    @LocalServerPort
    private int port;

    private final RestTemplate restTemplate = new RestTemplate();

    @Test
    void processEndpointReturnsOrchestratorResult() {
        String body = restTemplate.getForObject(
                "http://localhost:" + port + "/orchestrator/process",
                String.class);

        assertThat(body).contains("id_fsc", "+49170", "account", "ext_stammdaten");
    }
}

File: src/test/kotlin/com/example/dpop/orchestrator/ProcessControllerIntegrationTest.kt
package com.example.dpop.orchestrator

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.test.web.server.LocalServerPort
import org.springframework.test.context.ActiveProfiles
import org.springframework.web.client.RestTemplate

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
@ActiveProfiles("test")
class ProcessControllerIntegrationTest {

    @LocalServerPort
    private var port: Int = 0

    private val restTemplate = RestTemplate()

    @Test
    fun processEndpointReturnsOrchestratorResult() {
        val body = restTemplate.getForObject(
            "http://localhost:$port/orchestrator/process",
            String::class.java
        )

        assertThat(body).contains("id_fsc", "account", "ext_stammdaten")
    }
}